app/controllers/trainees/confirm_details_controller.rb

```diff
@@ -75,5 +76,10 @@ def mark_as_completed_params
 
       ActiveModel::Type::Boolean.new.cast(mark_as_completed_attributes)
     end
+
+    def check_training_route_permissions
+      return unless trainee.training_route.respond_to?("requires_#{trainee_section_key}?")
+      raise NotAuthorizedError unless trainee.training_route.public_send("requires_#{trainee_section_key}?")
```

Comment:
Can we add these into the pundit policies? We already use `authorize` in all of the actions and if this is an authorisation concern it would belong there.